### PR TITLE
BGP-5 [Backend-Frontend] Register Page

### PR DESCRIPTION
**Pull request by ZichanYang324**
# Description
This is the register page frontend and backend 

## Related PRS (if any):
after testing this, please check the #23 

## Main changes explained:
Add username, email, and password to register page in frontend, using MUI for the UI part, and add authmiddleware

## How to test:
1. check into the current branch
2. do `cd Server` and `npm i`, then `npm run dev`
3. do `cd Employee-Client` and `npm i`, then `npm run dev` to run this PR locally
4. add .env with `JWT_SECRET='abc123123123'` and `NODE_ENV=development`
5. add routes `.../register` then start testing the register function

## Screenshots or videos of changes or expecting result:

Screenshot 2024-02-25 at 13 12 51

Screenshot 2024-02-25 at 13 13 00

## Note:
Make sure to run both frontend and backend also add .env file

### CODE REVIEW

**Review by XiangweiChen — APPROVED** [review]
Followed the testing instructions, and the frontend/backend for register and login work as expected with the provided environment credentials keys in Server. Good job!

File: Employee-Client/src/App.jsx [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+import Dashboard from "./pages/dashboard";
 function App() {
-  return <></>;
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="login" element={<Login />} />
+        <Route path="register" element={<Register />} />
+        <Route path="dashboard" element={<Dashboard />} />
+      </Routes>
```
Comment (yuelikestocode) [inline Employee-Client/src/App.jsx:12]: Would it be better to move the routes to a dedicated component to leave `App.jsx` simple?

File: Server/controllers/userControllers.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+//     res.status(401);
+//     throw new Error('Invalid email or password');
+//   }
+// });
+
+// @desc    Register a new user
+// @route   POST /api/users
+// @access  Public
+const registerUser = asyncHandler(async (req, res) => {
+    console.log(req.body);
```
Comment (yuelikestocode) [inline Server/controllers/userControllers.js:33]: Could you remove console logs?

**Review by yuelikestocode — COMMENTED** [review]
What are the commented functions for?

**Review by christorange — APPROVED** [review]
LGTM!

**Comment by ZichanYang324** [comment]
@yuelikestocode I will modify that in my next PR, thank you for your reminder!

**Pull request merged**